src/DotVVM.Framework/Compilation/Binding/ExpressionBuildingVisitor.cs

```diff
@@ -329,6 +335,28 @@ protected override Expression VisitLambdaParameter(LambdaParameterBindingParserN
             return Expression.Parameter(parameterType, node.Name.ToDisplayString());
         }
 
+        private Expression CreateLambdaExpression(Expression body, ParameterExpression[] parameters)
+        {
+            // Check if we have type information available
+            if (expressionDepth == 1 && ExpectedType != null && ReflectionUtils.IsDelegate(ExpectedType))
```

Comment:
The `ExpectedType` is the top level expected type. However, the lambda may be used somewhere inside of the expression. ExpectedType might be string, while the expression is `MyMethod(x => Prop = x)` and it will lead to the same problem with `Action` and `Func`